File: AspNetMvcUnleashed/0672329980_source/Chapter05/CS/LSMvcApplication/LSMvcApplication/Controllers/HomeController.cs
using System.Linq;
using System.Web.Mvc;
using GenericRepository;
using LSMvcApplication.Models;

namespace LSMvcApplication.Controllers
{
    public class HomeController : Controller
    {
        private IGenericRepository _repository;


        public HomeController()
        {
            _repository = new LSGenericRepository(new DataModelDataContext());
        }



        //
        // GET: /Home/

        public ActionResult Index()
        {
            var result = (from p in _repository.List<Product>()
                         where p.Price < 50.00m
                         select p).ToList();
            return View(result);

            return View(_repository.List<Product>().ToList());
        }

        //
        // GET: /Home/Create

        public ActionResult Create()
        {
            return View();
        } 

        //
        // POST: /Home/Create

        [AcceptVerbs(HttpVerbs.Post)]
        public ActionResult Create([Bind(Exclude="Id")]Product productToCreate)
        {
            try
            {
                _repository.Create<Product>(productToCreate);
                return RedirectToAction("Index");
            }
            catch
            {
                return View();
            }
        }

        //
        // GET: /Home/Edit/5
 
        public ActionResult Edit(int id)
        {
            return View(_repository.Get<Product>(id));
        }

        //
        // POST: /Home/Edit/5

        [AcceptVerbs(HttpVerbs.Post)]
        public ActionResult Edit(Product productToEdit)
        {
            try
            {
                _repository.Edit<Product>(productToEdit); 
                return RedirectToAction("Index");
            }
            catch
            {
                return View();
            }
        }


        //
        // GET: /Home/Delete/5

        public ActionResult Delete(int id)
        {
            return View(_repository.Get<Product>(id));
        }

        //
        // POST: /Home/Delete/5

        [AcceptVerbs(HttpVerbs.Post)]
        public ActionResult Delete(Product productToDelete)
        {
            try
            {
                _repository.Delete<Product>(productToDelete);
                return RedirectToAction("Index");
            }
            catch
            {
                return View();
            }
        }



    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter05/VB/LSMvcApplication/LSMvcApplication/Controllers/HomeController.vb
Imports GenericRepository

Public Class HomeController
    Inherits Controller

    Private _repository As IGenericRepository


    Public Sub New()
        _repository = New LSGenericRepository(New DataModelDataContext())
    End Sub



    '
    ' GET: /Home/

    Public Function Index() As ActionResult
        Return View(_repository.List(Of Product)().ToList())
    End Function

    '
    ' GET: /Home/Create

    Public Function Create() As ActionResult
        Return View()
    End Function

    '
    ' POST: /Home/Create

    <AcceptVerbs(HttpVerbs.Post)> _
    Public Function Create(<Bind(Exclude:="Id")> ByVal productToCreate As Product) As ActionResult
        Try
            _repository.Create(Of Product)(productToCreate)
            Return RedirectToAction("Index")
        Catch
            Return View()
        End Try
    End Function



End Class
